### PR TITLE
Exclude joint filing units from the Delaware elderly or disabled exclusion if only one filer is eligible

### LINKED ISSUE #4237: Exclude joint filing units from the Delaware elderly or disabled exclusion if only one filer is eligible
Currently we assign a value if only one spouse is eligible in a joint unit 

Should pass this test:

```
- name: Tax unit with taxsimid 84350 in e21.its.csv and e21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        age: 63
        employment_income: 128010
        taxable_interest_income: 5505.0
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
        ma_covid_19_essential_employee_premium_pay_program: 0  # not in TAXSIM35
      person2:
        age: 63
        employment_income: 2010
        taxable_interest_income: 5505.0
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
        ma_covid_19_essential_employee_premium_pay_program: 0  # not in TAXSIM35
      person3:
        age: 11
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
        ma_covid_19_essential_employee_premium_pay_program: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2, person3]
        premium_tax_credit: 0  # not in TAXSIM35
        local_income_tax: 0  # not in TAXSIM35
        state_sales_tax: 0  # not in TAXSIM35
        ca_use_tax: 0  # not in TAXSIM35
        il_use_tax: 0  # not in TAXSIM35
        nm_2021_income_rebate: 0  # not in TAXSIM35
        nm_additional_2021_income_rebate: 0  # not in TAXSIM35
        nm_supplemental_2021_income_rebate: 0  # not in TAXSIM35
        ny_supplemental_eitc: 0  # not in TAXSIM35
        ok_use_tax: 0  # not in TAXSIM35
        pa_use_tax: 0  # not in TAXSIM35
        vt_renter_credit: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2, person3]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2, person3]
        state_fips: 10  # DE
  output:  # expected results from patched TAXSIM35 2024-03-07 version
    de_income_tax: 6585.82
```

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4237

### CODE REVIEW

File: policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py:1]: should this variable should be tax-unit-level now?

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
+    - Exclude joint filing units from the Delaware elderly or disabled exclusion if only one filer is eligible.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
    - Among joint filers, limit the Delaware elderly or disabled exclusion to filers with both head and spouse eligible.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py:1]: In the married filing separate case, the exclusion should apply to each person individually and I think if we change it to tax unit then we might attribute it to both even if only one person is eligible

File: policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py [hunk 2]
```diff
@@ -13,11 +13,29 @@ class de_elderly_or_disabled_income_exclusion(Variable):
     def formula(person, period, parameters):
         # First get their filing status.
         filing_status = person.tax_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py:15]: and use below
```suggestion
        tax_unit = person.tax_unit
        filing_status = tax_unit(
```

File: policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py [hunk 3]
```diff
@@ -13,11 +13,29 @@ class de_elderly_or_disabled_income_exclusion(Variable):
...
-            "state_filing_status_if_married_filing_separately_on_same_return",
+            "filing_status",
             period,
         )
         p = parameters(
             period
         ).gov.states.de.tax.income.subtractions.exclusions.elderly_or_disabled
 
-        return p.amount[filing_status]
+        # If filing jointly, both spouses have to be eligible,
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/de/tax/income/subtractions/exclusions/de_elderly_or_disabled_income_exclusion.py:23]:
```suggestion
        # If filing jointly, both spouses have to be eligible.
```

**Review by MaxGhenis — APPROVED** [review]
good to merge after this

**Pull request merged**